fangraphs/scores.py

```diff
@@ -515,3 +515,213 @@ def export(self):
+    def play_by_play(self):
+        """
+
+        :return:
+        :rtype: pd.DataFrame
+        """
+        return self._play_by_play
+
+    @play_by_play.setter
+    def play_by_play(self, value) -> None:
+        """
+
+        :param value:
+        """
+        if value is not None:
+            return
+
+        pbp_table = self._tables[4]
+        header_elems = pbp_table.select("thead > tr > th")
+        row_elems = pbp_table.select("tbody > tr")
+
+        dataframe = pd.DataFrame(
+            data=[
+                [e.text for e in r.select("td")]
+                for r in row_elems
+            ],
+            columns=[e.text for e in header_elems]
+        )
+        dataframe.drop(columns=dataframe.columns[-2:])
+
+        def pitch_sequence() -> list[list[str]]:
+            items = []
+            for row in row_elems:
+                elem = row.select_one("td:nth-child(7) a")
+                if elem is not None:
+                    pitches = elem.attrs.get("tooltip").split(",")
+                else:
+                    pitches = []
+                items.append(pitches)
+            return items
+
+        dataframe["Pitch Sequence"] = pitch_sequence()
+        dataframe["Player ID (Pitcher)"] = [
+            PID_POS_REGEX.search(
+                r.select_one("td:nth-child(1) a").attrs.get("href")
+            ).group(1) for r in row_elems
+        ]
+        dataframe["Player ID (PLayer)"] = [
+            PID_POS_REGEX.search(
+                r.select_one("td:nth-child(2) a").attrs.get("href")
+            ).group(1) for r in row_elems
+        ]
+
+        self._play_by_play = dataframe
+
+    @staticmethod
+    def _scrape_table(table):
+        """
+
+        :param table:
+        :type table: bs4.Tag
+        :return:
+        :rtype: pd.DataFrame
+        """
+        header_elems = table.select("thead > tr > th.rgHeader")
+        row_elems = table.select("tbody > tr")
+        dataframe = pd.DataFrame(
+            data=[
+                [e.text for e in r.select("td")]
+                for r in row_elems
+            ],
+            columns=[e.text for e in header_elems]
+        )
+
+        def playerids_positions() -> [list[str], list[str]]:
+            player_ids, positions, = [], []
+            for row in row_elems:
+                elem = row.select_one("td:nth-child(1)")
+                if elem.text != "Total":
+                    pid, position = PID_POS_REGEX.search(
+                        elem.select_one("a").attrs.get("href")
+                    ).groups()
+                else:
+                    pid, position = "", ""
+                player_ids.append(pid)
+                positions.append(position)
+            return player_ids, positions
+
+        dataframe["Player ID"], dataframe["Position"] = playerids_positions()
+
+        return dataframe
+
+    @property
+    def data_tables(self):
+        """
+
+        :return:
+        :rtype: dict[str, pd.DataFrame]
+        """
```

Comment:
Replace `:rtype:` docstring directive with function annotation (`->`). Type should be `NamedTuple[NamedTuple]` instead of `dict[str, pd.DataFrame]`.